### PR TITLE
add use_HF_format for export_compressed_model

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

feature

## Description

Huggingface GPTQ models are using a popular repo  https://github.com/qwopqwop200/GPTQ-for-LLaMa for compression.
I add an args in get_compressed_model(use_HF_fomat=True) to align with the popular one.

The main change for this arg is as below:

1: compression_dim: weight = 1, zeros = 0 and both are transposed.
2: zeros -= 1 before compression. Why we need it?
3: g_idx: use same number for one group instead of recording the channel order. 
4. parameter name changed, such as 'packed_weight' -> 'qweight'.
5. zeros is always needed even for sym.

## Expected Behavior & Potential Risk

UT pass

## How has this PR been tested?

local test

## Dependency Change?

N/A

### CODE REVIEW

File: docs/source/quantization_weight_only.md [hunk 1]
```diff
@@ -96,6 +96,14 @@ To support low memory inference, Neural Compressor implemented WeightOnlyLinear,
 |  compression_dtype  |       torch.int32       |  Data type for compressed dtype, select from [torch.int8\|16\|32\|64]   |
 |  compression_dim  |       1       |   0 means output channel while 1 means input channel   |
 |  scale_dtype  |       torch.float32       |  Data type for scale and bias   |
+|  use_HF_format  |     False       |  Whether to use the popular format present on HuggingFace hub   |
```
Comment (hshen14) [inline docs/source/quantization_weight_only.md:99]: use_hf_format?

File: docs/source/quantization_weight_only.md [hunk 2]
```diff
@@ -96,6 +96,14 @@ To support low memory inference, Neural Compressor implemented WeightOnlyLinear,
...
+
+**Note:** HuggingFace format is quite special, the main differences are as follows:
+
+> 1: Compression Dimension: weight = 1, zero = 0 and both are transposed.   
+> 2: Zero Point: zero_point-= 1 before compression. zero_point is always required even for sym.
```
Comment (hshen14) [inline docs/source/quantization_weight_only.md:104]: what's -= 1?

**Review by hshen14 — APPROVED** [review]

File: docs/source/quantization_weight_only.md (hunk 2 above)
Comment (xin3he) [reply docs/source/quantization_weight_only.md:104]: It's a popular behavior used in huggingface format, Weiwen and Thomas also cannot answer this question why we need it.

Comment (xin3he) [reply docs/source/quantization_weight_only.md:104]: It means zero_point = zero_point - 1

File: docs/source/quantization_weight_only.md (hunk 1 above)
Comment (xin3he) [reply docs/source/quantization_weight_only.md]: OK

**Review by hshen14 — APPROVED** [review]

**Review by changwangss — APPROVED** [review]

**Pull request merged**